applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h

```diff
@@ -797,38 +798,35 @@ class MPMStrategy
             }
 		}
 
-        //******************SEARCH FOR TRIANGLES************************
-        // Initialize shape function vector to be passed to PointLocator function
+        // Search background grid and make element active
         Vector N;
+        const int max_result = 1000;
 
-        if (m_GeometryElement == "Triangle")
+        #pragma omp parallel
         {
-            const int max_result = 1000;
-
             BinBasedFastPointLocator<TDim> SearchStructure(grid_model_part);
             SearchStructure.UpdateSearchDatabase();
 
-			#pragma omp parallel
-			{
-                typename BinBasedFastPointLocator<TDim>::ResultContainerType results(max_result);
+            typename BinBasedFastPointLocator<TDim>::ResultContainerType results(max_result);
 
-                #pragma omp for
-                for(int i = 0; i < static_cast<int>(mpm_model_part.Elements().size()); ++i){
+            #pragma omp for
+            for(int i = 0; i < static_cast<int>(mpm_model_part.Elements().size()); ++i){
 
-                    auto element_itr = mpm_model_part.Elements().begin() + i;
+                auto element_itr = mpm_model_part.Elements().begin() + i;
 
-                    array_1d<double,3> xg = element_itr -> GetValue(GAUSS_COORD);
-                    typename BinBasedFastPointLocator<TDim>::ResultIteratorType result_begin = results.begin();
+                array_1d<double,3> xg = element_itr -> GetValue(GAUSS_COORD);
+                typename BinBasedFastPointLocator<TDim>::ResultIteratorType result_begin = results.begin();
 
-                    Element::Pointer pelem;
+                Element::Pointer pelem;
 
-                    // FindPointOnMesh find the element in which a given point falls and the relative shape functions
-                    bool is_found = SearchStructure.FindPointOnMesh(xg, N, pelem, result_begin);
+                // FindPointOnMesh find the element in which a given point falls and the relative shape functions
+                bool is_found = SearchStructure.FindPointOnMesh(xg, N, pelem, result_begin);
 
-                    if (is_found == true)
+                if (is_found == true)
+                {
+                    pelem->Set(ACTIVE);
+                    if (m_GeometryElement == "Triangle")
                     {
-                        pelem->Set(ACTIVE);
-
                         element_itr->GetGeometry()(0) = pelem->GetGeometry()(0);
```

Comment:
I don't think this is a efficient way to replace the nodes

Reply:
@roigcarlo ?

Reply:
Thanks @loumalouomega. I will talk with @roigcarlo tomorrow!

Reply:
element_itr->GetGeometry() = pelem->GetGeometry()?

Reply:
Yep you can do that